src/mindmap.js

```diff
@@ -243,24 +264,38 @@ var FilesMindMap = {
 	},
 
 	/**
-	 * @param fileActions
 	 * @private
 	 */
-	registerFileActions: function(fileActions) {
+	registerFileActions: function () {
 		var mimes = this.getSupportedMimetypes(),
 			_self = this;
 
-		$.each(mimes, function(key, value) {
-			fileActions.registerAction({
-				name: 'Edit',
-				mime: value,
-				actionHandler: _.bind(_self._onEditorTrigger, _self),
-				permissions: OC.PERMISSION_READ,
-				icon: function () {
-					return OC.imagePath('core', 'actions/edit');
-				}
-			});
-			fileActions.setDefault(value, 'Edit');
+		$.each(mimes, function (key, value) {
+			registerFileAction(new FileAction({
+				id: 'file_mindmap-'+value,
+				displayName() {
+					return t('files_mindmap', 'Edit')
+				},
+				iconSvgInline: () => OC.imagePath('core', 'actions/edit'),
```

Comment:
Big doubt about that, but does it really matter as we won't see the action as it is the default one.